### PR TITLE
[CI] Fix GA oom issue

### PR DESCRIPTION
**Pull request by cfmcgrady**
### _Why are the changes needed?_

This PR aims to fix the OOM issue that happened on Github Action

### _How was this patch tested?_

Pass CI.

### CODE REVIEW

**Comment by cfmcgrady** [comment]
```
BatchesResourceSuite:
- open batch session
- get batch session list
- negative request
- batch sessions recovery
- get local log internal redirection
- delete batch internal redirection
- support to get the real client ip for http proxy use case
- expose the metrics with operation type and current state *** FAILED ***
  1 did not equal 0 (BatchesResourceSuite.scala:633)
```

the failed ut is a known issue #3114

**Comment by cfmcgrady** [comment]
```
22/07/23 07:00:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:01:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:02:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:03:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:04:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:05:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:06:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:07:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:08:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:09:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:10:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:11:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:12:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:13:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:14:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:15:20 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
22/07/23 07:16:23 INFO EngineServiceDiscovery: 2 connection(s) are active, delay shutdown
```
The unclosed sessions may block the leaked engine. In fact, these sessions should be closed by the `FeTServerEventHandler`, but I can't figure out the root cause, This PR closes these sessions after each Suite to prevent engine leak.

**Comment by cfmcgrady** [comment]
cc @pan3793 @yaooqinn @turboFei

**Review by pan3793 — APPROVED** [review]

**Review by turboFei — APPROVED** [review]

File: kyuubi-server/src/test/scala/org/apache/kyuubi/WithKyuubiServer.scala [hunk 1]
```diff
@@ -52,13 +54,23 @@ trait WithKyuubiServer extends KyuubiFunSuite {
...
     super.beforeAll()
   }
 
   override def afterAll(): Unit = {
+    server.frontendServices.foreach {
+      case frontend: AbstractFrontendService =>
+        val sessionManager = frontend.be.sessionManager.asInstanceOf[KyuubiSessionManager]
+        sessionManager.allSessions().foreach { session =>
+          logger.warn(s"found unclosed session ${session.handle}.")
+          sessionManager.closeSession(session.handle)
```
Comment (cxzl25) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/WithKyuubiServer.scala:70]: `closeSession` seems to throw exceptions in some cases, is it better to add a try catch?

**Review by cxzl25 — APPROVED** [review]

File: kyuubi-server/src/test/scala/org/apache/kyuubi/WithKyuubiServer.scala [hunk 2]
```diff
@@ -36,7 +40,7 @@ trait WithKyuubiServer extends KyuubiFunSuite {
   protected var server: KyuubiServer = _
 
   override def beforeAll(): Unit = {
-    conf.set(FRONTEND_PROTOCOLS, frontendProtocols.map(_.toString))
+    conf.setIfMissing(FRONTEND_PROTOCOLS, frontendProtocols.map(_.toString))
```
Comment (cfmcgrady) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/WithKyuubiServer.scala:43]: this is why the k8s integration tests hang. cc @turboFei

Comment (cfmcgrady) [reply kyuubi-server/src/test/scala/org/apache/kyuubi/WithKyuubiServer.scala:43]: another issue occurs after this one is fixed.

```
07:00:28.229 ERROR org.apache.kyuubi.server.metadata.MetadataManager: Error updating metadata for session 6654226d-c108-4669-98aa-f2c9535de379
org.apache.kyuubi.KyuubiException: A truncation error was encountered trying to shrink VARCHAR ' org.apache.spark.deploy.SparkSubmit --conf spark.kubernetes&' to length 1024.
	at org.apache.kyuubi.server.metadata.jdbc.JDBCMetadataStore.withUpdateCount(JDBCMetadataStore.scala:417) ~[kyuubi-server_2.12-1.6.0-SNAPSHOT.jar:1.6.0-SNAPSHOT]
	at org.apache.kyuubi.server.metadata.jdbc.JDBCMetadataStore.$anonfun$updateMetadata$7(JDBCMetadataStore.scala:274) ~[kyuubi-server_2.12-1.6.0-SNAPSHOT.jar:1.6.0-SNAPSHOT]
	at org.apache.kyuubi.server.metadata.jdbc.JDBCMetadataStore.$anonfun$updateMetadata$7$adapted(JDBCMetadataStore.scala:273) ~[kyuubi-server_2.12-1.6.0-SNAPSHOT.jar:1.6.0-SNAPSHOT]
	at org.apache.kyuubi.server.metadata.jdbc.JDBCMetadataStore.withConnection(JDBCMetadataStore.scala:445) ~[kyuubi-server_2.12-1.6.0-SNAPSHOT.jar:1.6.0-SNAPSHOT]
	at org.apache.kyuubi.server.metadata.jdbc.JDBCMetadataStore.updateMetadata(JDBCMetadataStore.scala:273) ~[kyuubi-server_2.12-1.6.0-SNAPSHOT.jar:1.6.0-SNAPSHOT]
	at org.apache.kyuubi.server.metadata.MetadataManager.updateMetadata(MetadataManager.scala:143) ~[kyuubi-server_2.12-1.6.0-SNAPSHOT.jar:1.6.0-SNAPSHOT]
	at org.apache.kyuubi.session.KyuubiSessionManager.updateMetadata(KyuubiSessionManager.scala:173) ~[kyuubi-server_2.12-1.6.0-SNAPSHOT.jar:1.6.0-SNAPSHOT]
	at org.apache.kyuubi.operation.BatchJobSubmission.updateBatchMetadata(BatchJobSubmission.scala:131) ~[kyuubi-server_2.12-1.6.0-SNAPSHOT.jar:1.6.0-SNAPSHOT]
	at org.apache.kyuubi.operation.BatchJobSubmission.submitAndMonitorBatchJob(BatchJobSubmission.scala:205) ~[kyuubi-server_2.12-1.6.0-SNAPSHOT.jar:1.6.0-SNAPSHOT]
	at org.apache.kyuubi.operation.BatchJobSubmission.$anonfun$runInternal$7(BatchJobSubmission.scala:173) ~[kyuubi-server_2.12-1.6.0-SNAPSHOT.jar:1.6.0-SNAPSHOT]
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23) ~[scala-library-2.12.15.jar:?]
	at scala.Option.getOrElse(Option.scala:189) ~[scala-library-2.12.15.jar:?]
	at org.apache.kyuubi.operation.BatchJobSubmission.$anonfun$runInternal$2(BatchJobSubmission.scala:173) ~[kyuubi-server_2.12-1.6.0-SNAPSHOT.jar:1.6.0-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) ~[?:?]
Caused by: java.sql.SQLDataException: A truncation error was encountered trying to shrink VARCHAR ' org.apache.spark.deploy.SparkSubmit --conf spark.kubernetes&' to length 1024.
	at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.jdbc.Util.generateCsSQLException(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.jdbc.EmbedConnection.handleException(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.jdbc.ConnectionChild.handleException(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeStatement(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeLargeUpdate(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeUpdate(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61) ~[HikariCP-4.0.3.jar:?]
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java) ~[HikariCP-4.0.3.jar:?]
	at org.apache.kyuubi.server.metadata.jdbc.JDBCMetadataStore.withUpdateCount(JDBCMetadataStore.scala:414) ~[kyuubi-server_2.12-1.6.0-SNAPSHOT.jar:1.6.0-SNAPSHOT]
	... 17 more
Caused by: org.apache.derby.iapi.error.StandardException: A truncation error was encountered trying to shrink VARCHAR ' org.apache.spark.deploy.SparkSubmit --conf spark.kubernetes&' to length 1024.
	at org.apache.derby.iapi.error.StandardException.newException(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.iapi.error.StandardException.newException(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.iapi.types.SQLChar.hasNonBlankChars(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.iapi.types.SQLVarchar.normalize(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.iapi.types.SQLVarchar.normalize(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.iapi.types.DataTypeDescriptor.normalize(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.sql.execute.NormalizeResultSet.normalizeColumn(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.sql.execute.NormalizeResultSet.normalizeRow(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.sql.execute.NormalizeResultSet.getNextRowCore(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.sql.execute.DMLWriteResultSet.getNextRowCore(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.sql.execute.UpdateResultSet.getNextRowCore(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.sql.execute.UpdateResultSet.collectAffectedRows(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.sql.execute.UpdateResultSet.open(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.sql.GenericPreparedStatement.executeStmt(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.sql.GenericPreparedStatement.execute(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeStatement(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeLargeUpdate(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at org.apache.derby.impl.jdbc.EmbedPreparedStatement.executeUpdate(Unknown Source) ~[derby-10.14.2.0.jar:?]
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61) ~[HikariCP-4.0.3.jar:?]
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java) ~[HikariCP-4.0.3.jar:?]
	at org.apache.kyuubi.server.metadata.jdbc.JDBCMetadataStore.withUpdateCount(JDBCMetadataStore.scala:414) ~[kyuubi-server_2.12-1.6.0-SNAPSHOT.jar:1.6.0-SNAPSHOT]
	... 17 more
```

File: kyuubi-server/src/test/scala/org/apache/kyuubi/WithKyuubiServer.scala (hunk 1 above)
Comment (cfmcgrady) [reply kyuubi-server/src/test/scala/org/apache/kyuubi/WithKyuubiServer.scala]: added.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**